src/plugins/dashboard/common/embeddable/embeddable_references.ts

```diff
@@ -62,7 +62,7 @@ export function extractPanelsReferences(
 
     const newPanel = convertPanelStateToSavedDashboardPanel(embeddable, panel.version);
     result.push({
-      panel: newPanel,
+      panel: { ...panel, ...newPanel },
```

Comment:
As I mentioned in https://github.com/elastic/kibana/issues/93772#issuecomment-792674769 I'd like suggest a more contained fix: 

The problem is clearly visibly here: 
https://github.com/elastic/kibana/blob/96c360b81d2c6800bd80703023386d0361f5cdef/src/plugins/dashboard/common/saved_dashboard_references.ts#L37

`extractPanelsReferences` expects a 7.3.0+ panel.
So I think we should put a `migratePanelsTo730()` in front of it. 
We do the same thing when we load a dashboard and get a state from the URL.

Reply:
I did notice this cast - but I figured it might be better to just leave the keys in and have the migrations run serverside later. Adding a migration here would effectively shift the full migration from server side to client side. If you think that is a less disruptive change - I trust your judgement. Thanks for looking into this!

Reply:
> Adding a migration here would effectively shift the full migration from server-side to client-side.

Not full, only pre 7.3.0

I basically suggesting adding existing that is always in play when you load a dashboard with panels in the state: https://github.com/elastic/kibana/blob/68ed17a486e4059fd3c903df7cb0a404f9d4872d/src/plugins/dashboard/public/application/lib/migrate_app_state.ts#L30-L83

That flow seems to be robustly tested and time-proof. So I suggesting replicating it with import saved object flow

Reply:
And there bunch of tests for supporting those URLs: https://github.com/elastic/kibana/blob/68ed17a486e4059fd3c903df7cb0a404f9d4872d/test/functional/apps/dashboard/bwc_shared_urls.ts
Would be nice to duplicate the same suite with import SO flow